client/extensions/woocommerce/state/sites/data/counts/selectors.js

```diff
@@ -1,61 +1,64 @@
-/**
- * External dependencies
- */
-import { get, reduce } from 'lodash';
-
 /**
  * Internal dependencies
  */
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { statusWaitingPayment, statusWaitingFulfillment } from 'woocommerce/lib/order-status';
 
+function getItems( state, siteId ) {
+	return state?.extensions?.woocommerce?.sites?.[ siteId ]?.data?.counts?.items ?? {};
+}
+
+function isLoading( state, siteId ) {
+	return state?.extensions?.woocommerce?.sites?.[ siteId ]?.data?.counts?.isLoading;
+}
+
 /**
  * @param {object} state Whole Redux state tree
  * @param {number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @returns {boolean} Whether the count data is already loaded for this site
  */
-export const areCountsLoaded = ( state, siteId = getSelectedSiteId( state ) ) => {
-	const isLoading = get( state, `extensions.woocommerce.sites[${ siteId }].data.counts.isLoading` );
-	return false === isLoading;
-};
+export function areCountsLoaded( state, siteId = getSelectedSiteId( state ) ) {
+	return isLoading( state, siteId ) === false;
+}
 
 /**
  * @param {object} state Whole Redux state tree
  * @param {number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @returns {boolean} Whether the count data is currently being retrieved from the server
  */
-export const areCountsLoading = ( state, siteId = getSelectedSiteId( state ) ) => {
-	const isLoading = get( state, `extensions.woocommerce.sites[${ siteId }].data.counts.isLoading` );
-	return true === isLoading;
-};
+export function areCountsLoading( state, siteId = getSelectedSiteId( state ) ) {
+	return isLoading( state, siteId ) === true;
+}
 
 /**
  * @param {object} state Whole Redux state tree
  * @param {number} [siteId] Site ID to check. If not provided, the Site ID selected in the UI will be used
  * @returns {number} The total number of products on this site
  */
-export const getCountProducts = ( state, siteId = getSelectedSiteId( state ) ) => {
-	const items = get( state, `extensions.woocommerce.sites[${ siteId }].data.counts.items`, {} );
-	return get( items, 'products.all', 0 );
-};
+export function getCountProducts( state, siteId = getSelectedSiteId( state ) ) {
+	const items = getItems( state, siteId );
+	const allProducts = items?.products?.all;
+	return allProducts !== undefined ? allProducts : 0;
```

Comment:
Judging by the inline documentation here, it seems like in the case of `allProducts === null`, we should be returning `0` as the selector will always return a number.

Since there are more of these cases in this PR, I'd suggest verifying all those assumptions with the folks from Woo.

Reply:
I changed them according to your suggestion, and am hoping one of the reviewers will be able to comment further.